inference-engine/src/mkldnn_plugin/nodes/mkldnn_lrn_node.cpp

```diff
@@ -1,58 +1,120 @@
-// Copyright (C) 2018-2020 Intel Corporation
+// Copyright (C) 2018-2021 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
 
 #include "mkldnn_lrn_node.h"
-
-#include <legacy/ie_layers.h>
 #include <string>
 #include <mkldnn_extension_utils.h>
+#include <ngraph/opsets/opset1.hpp>
 
-using namespace mkldnn;
 using namespace MKLDNNPlugin;
 using namespace InferenceEngine;
 
+bool MKLDNNLrnNode::isSupportedOperation(const std::shared_ptr<ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto lrn = std::dynamic_pointer_cast<const ngraph::opset1::LRN>(op);
+        if (!lrn) {
+            errorMessage = "Only opset1 LRN operation is supported";
+            return false;
+        }
+
+        const auto dataDims = lrn->get_input_shape(0);
+        if (dataDims.size() < 2 && dataDims.size() > 5) {
+            errorMessage = "Doesn't support 'data' input with rank: " + std::to_string(dataDims.size());
+            return false;
+        }
+        const auto axesNode = std::dynamic_pointer_cast<const ngraph::opset1::Constant>(lrn->get_input_node_shared_ptr(1));
+        if (!axesNode) {
+            errorMessage = "Only Constant operation on 'axis' input is supported";
+            return false;
+        }
+
+        const auto isSupportedAxes = [](const std::vector<int64_t> &axes, const size_t &dataRank) {
+            if (axes.size() == 1 && axes[0] == 1) {
+                return true;
+            } else {
+                std::vector<bool> norm(dataRank, false);
+                for (auto &axis : axes) {
+                    if (axis < 0) {
```

Comment:
Seems like we also need to check that `axis < dataRank`

Reply:
added check